designs/NuGet-ClientCertificates.md

```diff
@@ -0,0 +1,114 @@
+
+Added new configuration section **clientCertificates** which may have children of 2 types:
+
+**fromStorage** - item for certificate import from [certificate store](https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/working-with-certificates#certificate-stores). Internally uses [X509Certificate2Collection.Find](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2collection.find?view=netframework-4.8#System_Security_Cryptography_X509Certificates_X509Certificate2Collection_Find_System_Security_Cryptography_X509Certificates_X509FindType_System_Object_System_Boolean_) method.
+Can be configured with 4 Add item's
+
+- Optional Key="**StoreLocation**" Value="[possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storelocation?view=netframework-4.8#fields)".  
+Equals '**CurrentUser**' by default
+- Optional Key="**StoreName**" Value="[possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storename?view=netframework-4.8#fields)". 
+Equals '**My**' by default
+- Optional Key="**FindType**" Value="[possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509findtype?view=netframework-4.8#fields)". 
+Equals '**FindByThumbprint**' by default
+- Optional Key="**FindValue**" Value="The search criteria as an object"
+
+- **fromCert** - item for certificate import directly from file (DER encoded x.509, Base-64 encoded x.509, PKCS)
+Can be configured with 2 Add item's and item text body
+
+- Key="**Path**" Value="Absolute or relative path to certificate file". Relative path resolve have 2 stages: first relative to configuration file origin, second relative to current application directory.
+- Optional Key="**Password**" Value="Encrypted password". Password for certificate file. Encrypted in same manner as PackageSourceCredential password.
+- Items body can be filled with Base-64 encoded x.509 certificate. Start and end of text are trimmed from spaces and new line chars.
+
+## Configuration example
+
+```xml
+<configuration>
+...
+    <clientCertificates>	
+        <fromStorage>
+            <!-- Optional. CurrentUser by default -->
+            <add key="StoreLocation" value="CurrentUser" />
+            <!-- Optional. My by default -->
+            <add key="StoreName" value="My" />
+            <!-- Optional. FindByThumbprint by default -->
+            <add key="FindType" value="FindByThumbprint" />
+            <!-- Required. -->
+            <add key="FindValue" value="4894671ae5aa84840cc1079e89e82d426bc24ec6" />
+        </fromStorage>
+        <!-- Form 1 -->
+        <fromCert>
+            <!-- Absolute or relative path to certificate file. -->
+            <add key="Path" value=".\certificate.pfx" />
+            <!-- Encrypted password -->
+            <add key="Password" value="..." />
+        </fromCert>
```

Comment:
How about `<fromFile>`? Firstly this avoids overloading the same XML element for two different types of content, making writing validation or a schema file easier. Secondly, it's already in a section called `clientCertificates`, so `fromCert` feels a bit weird to me.

Reply:
First it was fromCert and fromFile but I decided to reduce code. If it is necessary will return additional specification.

Reply:
Split fromCert to fromFile and fromPEM